File: inject-java/src/test/groovy/io/micronaut/aop/simple/Invalid.java
package io.micronaut.aop.simple;

import io.micronaut.aop.Around;
import io.micronaut.context.annotation.Type;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.Target;

import static java.lang.annotation.RetentionPolicy.RUNTIME;

@Around
@Type(InvalidInterceptor.class)
@Documented
@Retention(RUNTIME)
@Target({ElementType.METHOD, ElementType.TYPE})
public @interface Invalid {
}

File: inject-kotlin/src/test/kotlin/io/micronaut/kotlin/processing/aop/simple/Invalid.kt
package io.micronaut.kotlin.processing.aop.simple

import io.micronaut.aop.Around
import io.micronaut.context.annotation.Type

@Around
@Type(InvalidInterceptor::class)
@MustBeDocumented
@Retention
@Target(
        AnnotationTarget.FUNCTION,
        AnnotationTarget.CLASS
)
annotation class Invalid
